aten/src/ATen/Context.cpp

```diff
@@ -311,6 +311,14 @@ const std::vector<at::QEngine>& Context::supportedQEngines() {
       engines.push_back(at::kFBGEMM);
     }
 #endif
+
+// The X86 qengine is available if and only if FBGEMM is available
+#ifdef USE_FBGEMM
+    if (fbgemm::fbgemmSupportedCPU()) {
+      engines.push_back(at::kX86);
```

Comment:
nit: can you put this in the same branch as L311

Reply:
is this related to `AT_MKLDNN_ENABLED` btw?

Reply:
> nit: can you put this in the same branch as L311

Sure. Done. I just thought it might be clearer if we separated them.

> is this related to `AT_MKLDNN_ENABLED` btw?

No. Since we only want to make it available when FBGEMM is available, no need to check MKLDNN here.